File: fsharp/DurableWorkflow/Workflow.fs
module Workflow

open Microsoft.Azure.WebJobs
open Microsoft.Extensions.Logging
open FSharp.Control.Tasks.V2.ContextInsensitive

let eventName = "Workflow"

[<FunctionName("Workflow")>]
let run
    ([<OrchestrationTrigger>] context : DurableOrchestrationContext)
    (logger : ILogger) =
    task {
        let input = context.GetInput<string>()
        sprintf "Starting workflow for %s" input |> logger.LogInformation

        do! context.WaitForExternalEvent(eventName)

        sprintf "Workflow for %s is stopping" input |> logger.LogInformation
    }

File: csharp/DurableWorkflow/Workflow.cs
using Microsoft.Azure.WebJobs;
using Microsoft.Extensions.Logging;
using System.Threading.Tasks;

namespace DurableWorkflow
{
    public static class Workflow
    {
        [FunctionName("Workflow")]
        public static async Task Run(
            [OrchestrationTrigger]DurableOrchestrationContext context,
            ILogger logger
            )
        {
            var input = context.GetInput<string>();
            logger.LogInformation($"Starting workflow for {input}");

            await context.WaitForExternalEvent(nameof(Workflow));

            logger.LogInformation($"Workflow for {input} is stopping");
        }
    }
}
